File: sdk/src/main/java/ru/usedesk/sdk/data/framework/api/standard/entity/request/InitChatRequest.java
package ru.usedesk.sdk.data.framework.api.standard.entity.request;

import com.google.gson.annotations.SerializedName;

import ru.usedesk.sdk.domain.entity.chat.Constants;

public class InitChatRequest extends BaseRequest {

    public static final String TYPE = "@@server/chat/INIT";

    @SerializedName(Constants.KEY_COMPANY_ID)
    private String companyId;

    private String url;
    private Payload payload;

    public InitChatRequest(String token, String companyId, String url) {
        super(TYPE, token);

        this.companyId = companyId;
        this.url = url;

        payload = new Payload();
    }

    public String getCompanyId() {
        return companyId;
    }

    public void setCompanyId(String companyId) {
        this.companyId = companyId;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Payload getPayload() {
        return payload;
    }

    private class Payload {

        private String sdk;

        private Payload() {
            this.sdk = Constants.VALUE_CURRENT_SDK;
        }

        public String getSdk() {
            return sdk;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/api/_entity/request/InitChatRequest.kt
package ru.usedesk.chat_sdk.data.repository.api._entity.request

import com.google.gson.annotations.SerializedName

class InitChatRequest(
        token: String?,
        @SerializedName(KEY_COMPANY_ID)
        val companyId: String,
        val url: String
) : BaseRequest(TYPE, token ?: "") {

    private val payload: Payload = Payload()

    class Payload {
        val sdk: String = VALUE_CURRENT_SDK
    }

    companion object {
        private const val TYPE = "@@server/chat/INIT"
        private const val KEY_COMPANY_ID = "company_id"
        private const val VALUE_CURRENT_SDK = "android"
    }
}